receiver/api/src/main/java/org/phenotips/integration/lims247/script/RemoteDataScriptService.java

```diff
@@ -83,10 +83,19 @@ public void save()
             if (isTrusted()) {
                 try {
                     @SuppressWarnings("deprecation")
-                    XWikiDocument doc = new XWikiDocument();
-                    doc.fromXML(context.getRequest().getInputStream());
-                    context.getWiki().saveDocument(doc, doc.getComment(), doc.isMinorEdit(), context);
-                    this.logger.debug("Imported [{}] from remote", doc.getDocumentReference());
+                    XWikiDocument xmlDoc = new XWikiDocument();
+                    xmlDoc.fromXML(context.getRequest().getInputStream());
+
+                    XWikiDocument existingDoc =
+                        context.getWiki().getDocument(xmlDoc.getDocumentReference(), context);
+
+                    if (!existingDoc.isNew()) {
+                        context.getWiki().deleteDocument(existingDoc, false, context);
```

Comment:
I would suggest adding a debug log before this line in case these deletes start failing so we have something in the logs to go off of

Reply:
Agreed, good idea